packages/viewer/src/lib/pins/model.ts

```diff
@@ -44,10 +50,34 @@ export function isTextPin(pin?: Pin): pin is TextPin {
   return pin?.type === 'text';
 }
 
+export interface PinStyleAttributes {
+  primaryColor?: Color.Color | string;
+  accentColor?: Color.Color | string;
+}
+
 export function isIconPin(pin?: Pin): pin is Pin {
   return pin?.type === 'icon';
 }
 
+function convertColorToString(color: Color.Color | string): string {
+  return typeof color === 'string' ? color : Color.toHexString(color);
+}
+export function getPinColors(pin?: Pin): {
+  primaryColor?: string;
+  accentColor?: string;
+} {
+  return {
+    primaryColor:
+      pin?.attributes?.style?.primaryColor == null
+        ? undefined
```

Comment:
Do we want to set a default value, or what will the pin look like if it's undefined?